Reland 267718@main with fixes.

sheeparegreat opened the pull request:
#### 5c073cc4a558bf6cd1d88cbfd123a202a27bb792
<pre>
Reland 267718@main with fixes.
<a href="https://bugs.webkit.org/show_bug.cgi?id=261051">https://bugs.webkit.org/show_bug.cgi?id=261051</a>
rdar://114846429

Reviewed by Don Olmstead and Alex Christensen.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/ShapeDetection/Interfaces/BarcodeFormatInterface.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPU.serialization.in: Added.
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUErrorFilter.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUIndexFormat.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUMapMode.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::scrollByUnits):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::executeScrollLineUp):
(WebCore::executeScrollLineDown):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::scrollViaNonPlatformEvent):
* Source/WebCore/platform/PlatformWheelEvent.h:
* Source/WebCore/platform/PlatformWheelEvent.serialization.in: Added.
* Source/WebCore/platform/ScrollTypes.h:
(WebCore::logicalToPhysical):
(WebCore::axisFromDirection):
* Source/WebCore/platform/ScrollTypes.serialization.in: Added.
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::ScrollableArea::scroll):
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::scrollbarForDirection const):
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::startTimerIfNeeded):
(WebCore::Scrollbar::pressedPartScrollDirection):
* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModel.serialization.in: Added.
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
(WebCore::ScrollAnimatorIOS::handleTouchEvent):
* Source/WebCore/platform/network/ProtectionSpaceBase.h:
* Source/WebCore/platform/network/ProtectionSpaceBase.serialization.in: Added.
* Source/WebCore/platform/network/soup/SoupNetworkProxySettings.h:
* Source/WebCore/platform/network/soup/SoupNetworkProxySettings.serialization.in: Added.
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/win/WebPopupMenuProxyWin.cpp:
(WebKit::WebPopupMenuProxyWin::onMouseWheel):
* Source/WebKit/UIProcess/win/WebView.cpp:
(WebKit::WebView::onHorizontalScroll):
(WebKit::WebView::onVerticalScroll):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::scrollBy):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::performNonEditingBehaviorForSelector):
* Source/WebKitLegacy/mac/WebView/WebFrameView.mm:
(-[WebFrameView _scrollToBeginningOfDocument]):
(-[WebFrameView _scrollToEndOfDocument]):
(-[WebFrameView _pageVertically:]):
(-[WebFrameView _pageHorizontally:]):
(-[WebFrameView _scrollLineVertically:]):
(-[WebFrameView _scrollLineHorizontally:]):

Canonical link: <a href="https://commits.webkit.org/267873@main">https://commits.webkit.org/267873@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a3805712f724cc96ebc390475a52e53d61b52f3f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17807 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18132 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18676 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19632 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16657 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18002 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21428 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18282 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18682 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18020 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18298 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15468 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20499 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/17955 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15526 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16222 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22775 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16542 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16390 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20640 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16962 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14355 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16067 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4270 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20424 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16808 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

donny-dont commented on Source/WebCore/platform/ScrollTypes.serialization.in:
```diff
@@ -0,0 +1,41 @@
+    ScrollRight,
+};
+
+enum class WebCore::ScrollbarOrientation : uint8_t {
```
Doesn't this also need `[CustomHeader] since there is not `ScrollbarOrientation.h`?

sheeparegreat replied:
Neither need [CustomHeader]. Removed it

donny-dont commented on Source/WebCore/platform/network/soup/SoupNetworkProxySettings.serialization.in:
```diff
@@ -0,0 +1,32 @@
+    Auto
+};
+
+#endif // USE(SOUP)
```
No newline

sheeparegreat replied:
Removed.

donny-dont commented on Source/WebKit/CMakeLists.txt:
```diff
@@ -687,6 +687,12 @@ set(WebCore_SERIALIZATION_IN_FILES
+    PlaybackSessionModel.serialization.in
+    ProtectionSpaceBase.serialization.in
+    ScrollTypes.serialization.in
+    SoupNetworkProxySettings.serialization.in
```
If its platform specific it shouldn't be in the `CMakeLists.txt`. Unfortunately there is no Soup specific CMake file in WebKit so you'll want to add these to `PlatformGTK.cmake ` and `PlatformWPE.cmake`.

You'll need to just do a `list(APPEND WebCore_SERIALIZATION_IN_FILES SoupNetworkProxySettings.serialization.in)` in each of those files.

sheeparegreat replied:
Moved SoupNetworkProxySettings.serialization.in as requested.

donny-dont reviewed (commented):
Couple nits. The Soup changes are dependent on https://github.com/WebKit/WebKit/pull/17594 so this will need to be rebased after that lands.

sheeparegreat committed:
Reland 267718@main with fixes.
https://bugs.webkit.org/show_bug.cgi?id=261051
rdar://114846429

Reviewed by Don Olmstead and Alex Christensen.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/ShapeDetection/Interfaces/BarcodeFormatInterface.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPU.serialization.in: Added.
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUErrorFilter.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUIndexFormat.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUMapMode.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::scrollByUnits):
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::executeScrollLineUp):
(WebCore::executeScrollLineDown):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::scrollViaNonPlatformEvent):
* Source/WebCore/platform/PlatformWheelEvent.h:
* Source/WebCore/platform/PlatformWheelEvent.serialization.in: Added.
* Source/WebCore/platform/ScrollTypes.h:
(WebCore::logicalToPhysical):
(WebCore::axisFromDirection):
* Source/WebCore/platform/ScrollTypes.serialization.in: Added.
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::ScrollableArea::scroll):
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::scrollbarForDirection const):
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::startTimerIfNeeded):
(WebCore::Scrollbar::pressedPartScrollDirection):
* Source/WebCore/platform/cocoa/PlaybackSessionModel.h:
* Source/WebCore/platform/cocoa/PlaybackSessionModel.serialization.in: Added.
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
(WebCore::ScrollAnimatorIOS::handleTouchEvent):
* Source/WebCore/platform/network/ProtectionSpaceBase.h:
* Source/WebCore/platform/network/ProtectionSpaceBase.serialization.in: Added.
* Source/WebCore/platform/network/soup/SoupNetworkProxySettings.h:
* Source/WebCore/platform/network/soup/SoupNetworkProxySettings.serialization.in: Added.
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/win/WebPopupMenuProxyWin.cpp:
(WebKit::WebPopupMenuProxyWin::onMouseWheel):
* Source/WebKit/UIProcess/win/WebView.cpp:
(WebKit::WebView::onHorizontalScroll):
(WebKit::WebView::onVerticalScroll):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::scrollBy):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::performNonEditingBehaviorForSelector):
* Source/WebKitLegacy/mac/WebView/WebFrameView.mm:
(-[WebFrameView _scrollToBeginningOfDocument]):
(-[WebFrameView _scrollToEndOfDocument]):
(-[WebFrameView _pageVertically:]):
(-[WebFrameView _pageHorizontally:]):
(-[WebFrameView _scrollLineVertically:]):
(-[WebFrameView _scrollLineHorizontally:]):

Canonical link: https://commits.webkit.org/267873@main
